app/models/katello/root_repository.rb

```diff
@@ -187,10 +187,16 @@ def ensure_ostree_repo_protected
 
     def ensure_valid_collection_attributes
       errors.add(:base, _("URL needs to have a trailing /")) if !url.blank? && url[-1] != '/'
+      return unless ansible_collection_requirements
       begin
-        YAML.safe_load(ansible_collection_requirements) if ansible_collection_requirements
+        requirements = YAML.safe_load(ansible_collection_requirements)
+        if requirements.is_a?(Hash)
+          errors.add(:base,  _("Requirements yaml should have a 'collections' key")) unless requirements.key?('collections')
+        else
+          errors.add(:base,  _('Requirements yaml should be a key-value pair structure.'))
```

Comment:
I tried JSON (`{"collections": "theforeman.foreman"}`) and got an error.  I think if we mention that it must be "key-value", then other types of key-value data structures should work.

```
{"pulp_tasks"=>
  [{"pulp_href"=>"/pulp/api/v3/tasks/553cb506-a3b5-4ae3-969b-fb5c4e6e449b/",
    "pulp_created"=>"2021-08-16T20:40:38.454+00:00",
    "state"=>"failed",
    "name"=>"pulp_ansible.app.tasks.collections.sync",
    "logging_cid"=>"a8464ef0-0351-4457-aec6-833386db9cc6",
    "started_at"=>"2021-08-16T20:40:38.511+00:00",
    "finished_at"=>"2021-08-16T20:40:38.826+00:00",
    "error"=>
     {"traceback"=>
       "  File \"/usr/lib/python3.6/site-packages/pulpcore/tasking/pulpcore_worker.py\", line 272, in _perform_task\n" +
       "    result = func(*args, **kwargs)\n" +
       "  File \"/usr/local/lib/python3.6/site-packages/pulp_ansible/app/tasks/collections.py\", line 130, in sync\n" +
       "    repo_version = d_version.create()\n" +
       "  File \"/usr/lib/python3.6/site-packages/pulpcore/plugin/stages/declarative_version.py\", line 151, in create\n" +
       "    loop.run_until_complete(pipeline)\n" +
       "  File \"/usr/lib64/python3.6/asyncio/base_events.py\", line 484, in run_until_complete\n" +
       "    return future.result()\n" +
       "  File \"/usr/lib/python3.6/site-packages/pulpcore/plugin/stages/api.py\", line 225, in create_pipeline\n" +
       "    await asyncio.gather(*futures)\n" +
       "  File \"/usr/lib/python3.6/site-packages/pulpcore/plugin/stages/api.py\", line 43, in __call__\n" +
       "    await self.run()\n" +
       "  File \"/usr/local/lib/python3.6/site-packages/pulp_ansible/app/tasks/collections.py\", line 778, in run\n" +
       "    await asyncio.gather(*tasks)\n" +
       "  File \"/usr/local/lib/python3.6/site-packages/pulp_ansible/app/tasks/collections.py\", line 587, in _fetch_collection_metadata\n" +
       "    namespace, name = requirements_entry.name.split(\".\")\n",
      "description"=>"not enough values to unpack (expected 2, got 1)"},
    "worker"=>"/pulp/api/v3/workers/a591b806-ac26-46d6-95e1-cdb1c2c66564/",
    "child_tasks"=>[],
    "progress_reports"=>
     [{"message"=>"no_change: Checking if remote changed since last sync.",
       "code"=>"sync.no_change",
       "state"=>"completed",
       "done"=>0},
      {"message"=>"Downloading Artifacts",
       "code"=>"sync.downloading.artifacts",
       "state"=>"canceled",
       "done"=>0},
      {"message"=>"Downloading Artifacts",
       "code"=>"sync.downloading.artifacts",
       "state"=>"canceled",
       "done"=>0},
      {"message"=>"Associating Content",
       "code"=>"associating.content",
       "state"=>"canceled",
       "done"=>0}],
    "created_resources"=>[],
    "reserved_resources_record"=>
     ["/pulp/api/v3/remotes/ansible/collection/41211e3c-e895-49e2-8146-962389697c1b/",
      "/pulp/api/v3/repositories/ansible/ansible/74cadfbf-87c6-4381-b03d-8bd9d7735fbb/"]}],
 "create_version"=>true,
 "task_groups"=>[],
 "poll_attempts"=>{"total"=>1, "failed"=>1}}

Exception:

Katello::Errors::Pulp3Error: not enough values to unpack (expected 2, got 1)
```

Reply:
we suggest in a way that its supposed to be yaml, and you only get that error if you pass valid yaml that isn't a hash.... but apparently json is kinda/sorta a subset of yaml, but in a way that i guess pulp does not like?  

I'm not sure what to say here, as it has to be a dict, map, hash, key/value pair, whatever you want to call it, but that form of yaml just doesn't work with pulp.....  Maybe we can suggest to look at the documentation for requirements.yaml as part of this error?

Reply:
> Maybe we can suggest to look at the documentation for requirements.yaml as part of this error?

I think this is the safest way to go.  I didn't realize that valid yaml could be anything besides a hash.

Reply:
opened https://pulp.plan.io/issues/9250